mods/beds/functions.lua

```diff
@@ -274,7 +275,10 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 	local last_player_in_bed = get_player_in_bed_count()
 
 	if fields.quit or fields.leave then
-		lay_down(player, nil, nil, false)
+		if beds.player[player:get_player_name()] then
+			-- make player stand up only if they were in bed
+			lay_down(player, nil, nil, false)
+		end
```

Comment:
I'd prefer to see this check in `lay_down`. That would properly guard from other edge-cases as well. Throwing an error is not necessary the best idea for servers. If there's no need to perform a "stand up", just return silently.

Reply:
How about keeping this guard, and logging a warning instead of throwing an error?

Reply:
I guess that might be okay too

Reply:
Moved the check to `lay_down`. Haven't added a warning as it's only used internally anyways.